pydm/widgets/eventplot.py

```diff
@@ -105,6 +105,9 @@ def receiveValue(self, new_data):
             return
         if self.x_idx is None or self.y_idx is None:
             return
+        if not isinstance(self.x_idx, int) or not isinstance(self.y_idx, int):
+            self.x_idx = int(self.x_idx)
+            self.y_idx = int(self.y_idx)
```

Comment:
How late does this need to be? Waiting till the first value is very late. Is doing this in `__init__` too early? Or does doing it on `__init__` raise an exception in the designer or something?

Reply:
It looks like a `ValueError` from designer trying to convert a macro string to an int is probably the reason it wasn't put in `__init__`? It's fine at runtime when the macro substitution actually happens. I think either something like

```
from ..utilities import is_qt_designer

if not is_qt_designer():
    self.x_idx = int(x_idx)
    self.y_idx = int(y_idx)
```
Or just a try/except catching the `ValueError` would work to move it into `__init__`. Maybe there's something more clever I'm not thinking of too.

Reply:
Yes that is exactly that problem. I did not know about that is_qt_designer functionality, this would be a better solution than mine indeed! Let me try it and I might update the PR accordingly.